src/math/trig.c

```diff
@@ -94,9 +97,197 @@ void cordic_sin_cos(int32_t th_rad_fxp, cordic_cfg type, int32_t *sign, int32_t
 		xtmp = *xn >> (b_idx + 1);
 		ytmp = *b_yn >> (b_idx + 1);
 	}
-	/* Q2.30 format */
+	/* Q2.30 format -sine, cosine*/
 	*th_cdc_fxp = th_rad_fxp;
+}
+
+/**
+ * \CORDIC-based approximation for inverse cosine
+ * Arguments	: int32_t cosvalue
+ *		  int16_t numiters
+ *		  const int32_t LUT[CORDIC_30B_ITABLE_SIZE]
+ * Return Type	: int32_t
+ */
+int32_t is_scalar_cordic_acos(int32_t cosvalue, int16_t numiters,
+			      const int32_t *LUT)
+{
+	int32_t b_cosvalue;
+	int32_t b_i;
+	int32_t i;
+	int32_t x = 0;
+	int32_t xdshift;
+	int32_t xshift;
+	int32_t y = 0;
+	int32_t ydshift;
+	int32_t yshift;
+	int32_t z = 0;
+	int32_t sign;
+	int16_t j;
+	int16_t k;
 
+	b_cosvalue = (cosvalue >> 1);
+	/* Initialize the variables for the cordic iteration
+	 * angles less than pi/4, we initialize (x,y) along the x-axis.
+	 * angles greater than or equal to pi/4, we initialize (x,y)
+	 * along the y-axis. This improves the accuracy of the algorithm
+	 * near the edge of the domain of convergence
+	 */
+	if (b_cosvalue < cord_arcsincos_q28fl) {
```

Comment:
Is this correct? cosvalue is Q1.31, right-shifted by 1 so b_cosvalue is Q2.29. However you compare it with Q4.28.

Reply:
Maybe rename cord_arcsincos_q28fl as cord_arcsincos_sqrt2over2_q28fl, or cord_arcsincos_oneoversqrttwo_q28fl?
Similar to cord_arcsincos_q30fl, maybe rename as cord_arcsincos_one_q30fl

Reply:
Reverted ,It was mistake. saturated_unary_minus is  Q2.30 [-1 to 1] range . 
fi(-1:0.1:1,1,32)
        WordLength: 32
        FractionLength:30

Reply:
Q2.30 --( >>1 )--> Q3.29, it still should not compare to Q4.28?

Reply:
I don't understand why ? Please take a look for below example
```
b_cosvalue = (cosvalue >> 1);
	if (b_cosvalue < cord_arcsincos_q28fl) {
		x = 0;
		y = cord_arcsincos_q30fl;
		z = PI_DIV2_Q3_29;
	} else {
		x = cord_arcsincos_q30fl;
		y = 0;
		z = 0;
	}
```
example with data
```
|Variable		| Value		|log2 (Value)	   |
+-----------------------+---------------+------------------+
|cord_arcsincos_q28fl	|189812531	|27.499999998111221|
|b_cosvalue		|161061273	|27.263034400459336|
|cosvalue		|322122547	|28.263034404938050|
```

Reply:
The comment about cord_arcsincos_q28fl confuses me. Please see my question there.